packages/cli/commands/project/upload.js

```diff
@@ -100,10 +102,14 @@ exports.handler = async options => {
 
   trackCommandUsage('project-upload', { projectPath }, accountId);
 
-  const cwd = projectPath ? path.resolve(getCwd(), projectPath) : getCwd();
-  const projectConfig = await getProjectConfig(cwd);
+  const projectDir = projectPath
+    ? path.resolve(getCwd(), projectPath)
+    : getCwd();
+  const projectConfig = await getProjectConfig(projectDir);
+
+  validateProjectConfig(projectConfig, projectDir);
 
-  validateProjectConfig(projectConfig);
+  await ensureProjectExists(accountId, projectConfig.name);
```

Comment:
Are we sure that accountId exists here? I was seeing an issue when testing this (sometime last week) where it was logging `The project <Project name> does not exist in undefined. Would you like to create it?`

Reply:
So yeah, I believe accountId should always exist here. If it doesn't https://github.com/HubSpot/hubspot-cli/pull/564/files#diff-1989a521c805539a7ae5e60fa3f7b0f79d83106bbb225045cf33f831abf97757R98 should be ending the process